yaksh/views.py

```diff
@@ -2591,14 +2618,14 @@ def design_module(request, module_id, course_id=None):
                 for order, value in enumerate(add_values, start_val):
                     learning_id, type = value.split(":")
                     if type == "quiz":
-                        learning_unit = LearningUnit.objects.create(
+                        unit, status = LearningUnit.objects.get_or_create(
                             order=order, quiz_id=learning_id,
                             type=type)
                     else:
-                        learning_unit = LearningUnit.objects.create(
+                        unit = LearningUnit.objects.get_or_create(
```

Comment:
```unit, status``` should come  as returns tuple
and unit you need to pass to ```...learning_unit.add(*units_list)```, will fail if tuple is passed.